sdcm/logcollector.py

```diff
@@ -948,14 +948,36 @@ def collect_logs(self, local_search_path: Optional[str] = None) -> Optional[str]
 
             if not os.listdir(self.local_dir):
                 LOGGER.warning('Nothing found')
-                return None
+                return []
+
+        if self.params.get("collect_single_archive"):
+            s3_links = self.create_single_archive_and_upload()
+        else:
+            s3_links = self.create_achive_per_file_and_upload()
 
-        final_archive = self.archive_dir_to_tarfile(self.local_dir)
+        return s3_links
+
+    def create_single_archive_and_upload(self) -> list[str]:
+        final_archive = self.archive_to_tarfile(self.local_dir)
 
         s3_link = upload_archive_to_s3(final_archive, f"{self.test_id}/{self.current_run}")
         remove_files(self.local_dir)
         remove_files(final_archive)
-        return s3_link
+        return [s3_link]
+
+    def create_achive_per_file_and_upload(self) -> list[str]:
+        s3_links = []
+        for root, _, files in os.walk(self.local_dir):
+            for f in files:
```

Comment:
In second commit you fix pylint issue implemented here.
So, please, just do not implement that pylint issue.
Do not skip, but use some proper name for loop var.
For example:
```
for current_file in files:
    ...
```

Reply:
fixed